File: easy/onboarding/F#/onboarding.fs
(* CodinGame planet is being attacked by slimy insectoid aliens. *)
(* <--- *)
(* Hint: To protect the planet, you can implement the pseudo-code provided in the statement, below the player. *)
open System

(* game loop *)
while true do
    let enemy1 = Console.In.ReadLine() (* name of enemy 1 *)
    let dist1 = int (Console.In.ReadLine()) (* distance to enemy 1 *)
    let enemy2 = Console.In.ReadLine() (* name of enemy 2 *)
    let dist2 = int (Console.In.ReadLine()) (* distance to enemy 2 *)
    
    (* Determine which enemy is closer and print its name *)
    printfn "%s" (if dist1 < dist2 then enemy1 else enemy2)

File: easy/onboarding/VB.NET/onboarding.vb
Module Player
' CodinGame planet is being attacked by slimy insectoid aliens.
' <---
' Hint: To protect the planet, you can implement the pseudo-code provided in the statement, below the player.

    Sub Main ()
        
        ' game loop
        While True
            Dim enemy1 as String = Console.ReadLine() ' name of enemy 1
            Dim dist1 as Integer = Integer.Parse(Console.ReadLine()) ' distance to enemy 1
            Dim enemy2 as String = Console.ReadLine() ' name of enemy 2
            Dim dist2 as Integer = Integer.Parse(Console.ReadLine()) ' distance to enemy 2

            ' Write an action using Console.WriteLine()
            ' To debug: Console.Error.WriteLine("Debug messages...")

            ' Determine which enemy is closer and print its name
            Console.WriteLine(If(dist1 < dist2, enemy1, enemy2))
        End While
    End Sub
End Module